### ISSUE TITLE
Transfer learning freeze layers

### ISSUE THREAD

**Issue by Igal20** [facebookresearch/maskrcnn-benchmark#201]
## ❓ Questions and Help
Hello,
I would like to perform transfer learning and that is bu freezing part of the layers.
For example if I use R-50-FPN net I know that Resnet-50 has 50 layer so Iwan to be able to freeze first 40 layers and train only latest 10.
I tried to do it here:
```
# Add StopGrad at a specified stage so the bottom layers are frozen
_C.MODEL.BACKBONE.FREEZE_CONV_BODY_AT = 2
```
but there are only 4 layer.
Am I missing something?
How can I do that?

**Comment by fmassa** [comment]
Hi,

By default and for simplicity, we freeze entire resnet stages, see https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/modeling/backbone/resnet.py#L112-L113 for an example.

You can modify that part of the code to better suit your needs, but if you are ok training the last 9 layers (instead of 10) you can just set `FREEZE_CONV_BODY_AT = 3`, and you'll train the [last two resnet stages](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/modeling/backbone/resnet.py#L50), which sums up to 9 layers.

**Comment by guanbin1994** [comment]
> FREEZE_CONV_BODY_AT = 3

what is the difference when FREEZE_CONV_BODY_AT set to 3 or 4 or 5 
I do not know FREEZE_CONV_BODY_AT actually doing what?

**Comment by fmassa** [comment]
If you check the link I sent, it is actually setting `requires_grad = False` to the parameters.
This means that gradients won't be computed for those layers, and they will be kept frozen / fixed during training.

**Comment by fmassa** [comment]
As you can see from [here](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/modeling/backbone/resnet.py#L111), if you want to only train starting from stage 4, you should set `FREEZE_CONV_BODY_AT = 4`

**Comment by guanbin1994** [comment]
> As you can see from [here](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/modeling/backbone/resnet.py#L111), if you want to only train starting from stage 4, you should set `FREEZE_CONV_BODY_AT = 4`

I remember training last four stage means training starting from stage 2 ?

**Comment by fmassa** [comment]
If you want to train the last 4 stages, then yes, you should start from stage 2, as is the default in the config.
